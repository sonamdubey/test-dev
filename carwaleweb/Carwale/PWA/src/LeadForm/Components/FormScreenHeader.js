import React, { PureComponent } from "react";
import PropTypes from "prop-types";

class FormScreenHeader extends PureComponent {
  render() {
    return (
      <div>
        {this.props.isCrossSell ? (
          <div className="dealer-detail">
            <p className="dealer-detail__heading">
              <span className="text-unbold">
                Get assistance from{" "}
                <span className="text-bold">{this.props.campaignName}</span> for
              </span>
            </p>
            <div className="car-img-fold">
              <img
                className="pq-m-car-img"
                src={this.props.imageUrl}
                alt={this.props.fullCarName}
              />
              <p className="pq-m-car-name">{this.props.fullCarName}</p>
            </div>
          </div>
        ) : (
          <div className="dealer-detail">
            <p className="dealer-detail__heading">{this.props.campaignName}</p>
            <p className="dealer-detail__subheading">
              Provide your contact details for further communication related to
              Test Drive, EMI options, Offers &amp; Exchange Benefits
            </p>
          </div>
        )}
      </div>
    );
  }
}

FormScreenHeader.propTypes = {
  isCrossSell: PropTypes.bool,
  campaignName: PropTypes.string,
  modelName: PropTypes.string,
  imageUrl: PropTypes.string
};

export default FormScreenHeader;
