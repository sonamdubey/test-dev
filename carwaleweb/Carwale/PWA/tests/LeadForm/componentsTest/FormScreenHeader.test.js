import React from "react";
import FormScreenHeader from "../../../src/LeadForm/Components/FormScreenHeader";

test("Component FormScreenHeader Snapshot RenderedTree (1)", () => {
  let props = {
    isCrossSell: true,
    campaignName: "Shivam Autozone",
    imageUrl:
      "https://imgd.aeplcdn.com/310x174/cw/ec/26860/Maruti-Suzuki-Dzire-Exterior-118637.jpg?wm=0",
    fullCarName: "Maruti Suzuki Dzire"
  };
  let wrapper = shallow(<FormScreenHeader {...props} />);
  expect(wrapper).toMatchSnapshot();
});

test("Component FormScreenHeader Snapshot RenderedTree (2)", () => {
  let props = {
    isCrossSell: false,
    imageUrl: "",
    campaignName: "Shivam Autozone",
    fullCarName: ""
  };

  let wrapper = shallow(<FormScreenHeader {...props} />);
  expect(wrapper).toMatchSnapshot();
});
