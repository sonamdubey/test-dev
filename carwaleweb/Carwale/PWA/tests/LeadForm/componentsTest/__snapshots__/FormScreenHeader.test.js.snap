// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component FormScreenHeader Snapshot RenderedTree (1) 1`] = `
<div>
  <div
    className="dealer-detail"
  >
    <p
      className="dealer-detail__heading"
    >
      <span
        className="text-unbold"
      >
        Get assistance from
         
        <span
          className="text-bold"
        >
          Shivam Autozone
        </span>
         for
      </span>
    </p>
    <div
      className="car-img-fold"
    >
      <img
        alt="Maruti Suzuki Dzire"
        className="pq-m-car-img"
        src="https://imgd.aeplcdn.com/310x174/cw/ec/26860/Maruti-Suzuki-Dzire-Exterior-118637.jpg?wm=0"
      />
      <p
        className="pq-m-car-name"
      >
        Maruti Suzuki Dzire
      </p>
    </div>
  </div>
</div>
`;

exports[`Component FormScreenHeader Snapshot RenderedTree (2) 1`] = `
<div>
  <div
    className="dealer-detail"
  >
    <p
      className="dealer-detail__heading"
    >
      Shivam Autozone
    </p>
    <p
      className="dealer-detail__subheading"
    >
      Provide your contact details for further communication related to Test Drive, EMI options, Offers & Exchange Benefits
    </p>
  </div>
</div>
`;
